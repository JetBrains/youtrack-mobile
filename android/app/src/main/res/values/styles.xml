<resources>

    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
        <!-- Customize your theme here. -->
        <item name="android:windowBackground">@drawable/background_splash</item>
    </style>

</resources>
