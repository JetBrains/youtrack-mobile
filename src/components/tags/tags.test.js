import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Tags from './tags';
import styles from './tags.styles';

describe('<Tags/>', () => {

  let wrapper;
  let instance;
  let tagQueryMock;
  let onPressMock;
  let tagMock;

  beforeEach(() => {
    tagQueryMock = 'tag: foo';
    onPressMock = jest.fn();
    tagMock = {
      color: {id: 1},
      query: tagQueryMock
    };
    wrapper = doShallow([tagMock], onPressMock);
    instance = wrapper.instance();
  });


  describe('Render', () => {
    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component', () => {
      expect(findByTestId('tagsList')).toHaveLength(1);
    });
  });


  describe('_showActions', () => {
    it('should invoke prop`s `onTagPress` fn', async () => {
      instance._getSelectedActions = jest.fn(() => instance._getContextActions(tagMock)[0]);
      instance.forceUpdate();

      await instance._showActions(tagMock);

      expect(instance.props.onTagPress).toHaveBeenCalledWith(tagMock.query);
    });
  });


  describe('_toggleShowAll', () => {
    it('should toggle `showAllTags` state', () => {
      instance._toggleShowAll();
      expect(instance.state.showAllTags).toEqual(true);

      instance._toggleShowAll();
      expect(instance.state.showAllTags).toEqual(false);
    });
  });


  describe('isDefaultColorCoding', () => {
    it('should return TRUE', () => {
      expect(
        instance.isDefaultColorCoding({
          color: {id: '0'}
        })
      ).toEqual(styles.tagNoColor);
    });

    it('should return NULL', () => {
      expect(
        instance.isDefaultColorCoding(tagMock)
      ).toEqual(null);
    });
  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(tags = [], onTagPress = () => {}, inline = false) {
    return shallow(
      <Tags
        tags={tags}
        onTagPress={onTagPress}
        inline={inline}
      />
    );
  }
});
