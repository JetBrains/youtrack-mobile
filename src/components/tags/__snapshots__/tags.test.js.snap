// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags/> Render should match a snapshot 1`] = `
<View
  style={
    Array [
      undefined,
      null,
      undefined,
    ]
  }
  testID="tagsList"
>
  <TouchableOpacity
    activeOpacity={0.2}
    key="undefined_button"
    onPress={[Function]}
    testID="tagsListTag"
  >
    <ColorField
      color={
        Object {
          "id": 1,
        }
      }
      defaultColorCoding={null}
      fullText={true}
      style={
        Array [
          Array [
            undefined,
            null,
          ],
          Object {
            "marginLeft": 0,
          },
        ]
      }
      testID="tagColor"
    />
  </TouchableOpacity>
</View>
`;
